src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SQLDNSCache.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+
+namespace Microsoft.Data.SqlClient
+{
+    internal class SQLDNSCache
+    {
+        private static readonly SQLDNSCache _SQLDNSCache = new SQLDNSCache();
```

Comment:
When will this cache clear at all?
Can we provide a way to clear DNS Cache like we have "ClearAllPools" public API on `SqlConnection`?

Reply:
Currently, there is no control of clearing all items in the cache. Neither the client or the server has control of that. We will remove specific items from cache when the server tells us to do so but it should not impact other items in the cache. Only when the application is stopped, the cache will be recycled.

Reply:
I don't know if there would be users who would like to clear this up and control memory allocations.
@David-Engel any thoughts?

Reply:
If we do a little math, estimate 60 bytes of data per entry (ex: 12345678901234567890.database.windows.net 111.111.111.111), 3(?) entries per DB (GW, CR, TR?), an application that makes 1000 connections to different databases, and we get 176kB. Maybe double that for overhead of the ConcurrentDictionary. It really should never grow beyond the number of different databases the application connects to. It doesn't seem like applications should need to control it.

Reply:
Based on the previous discussion with server team, this feature is boxed inside the driver and is not exposed to the client application. I think the cache should hold the records during the application's lifetime. For now, if we connect to the server endpoint and receive the flag from server saying that we need to remove its entry, the driver will remove that particular one from the cache. But for other records for other servers / FQDNs, they will still in the cache. They can still be used by other connections while the application is running. It's a bit tricky to find the proper timing to clean up the whole DNS cache while the application is running if it is not controlled by the client.